accepted/00053-disallow-downloading-when-patching.md

```diff
@@ -0,0 +1,61 @@
+- Feature Name: disallow-downloading-when-patching
+- Start Date: 29-03-2919
+- RFC PR: https://github.com/uyuni-project/uyuni-rfc/pull/6
+
+# Summary
+Optionally fail package installations and patch applications if they require any downloading.
+
+This is limited to minions on SUSE-based distros.
+
+# Motivation
+Minions with limited connectivity options might have little bandwidth available or bandwidth usage could come with a high cost (eg. in remote areas). In those cases we want users to be fully in control of when downloading happens, eg., to schedule that during off-peak hours (see https://github.com/uyuni-project/uyuni-rfc/pull/6). This RFC is about giving users the possibility to prevent package managers' default behavior of downloading at installation time.
+
+Package installation and patch application Actions should thus fail in case packages were not previously downloaded (if so configured).
+
+# Detailed design
+- two new functions are implemented in [zypper's execution module](https://github.com/openSUSE/salt/blob/openSUSE-2018.3.0/salt/modules/zypper.py). Given a list of packages or patches, return `True` if any downloading is required
+
+Those functions can leverage `zypper`'s estimation of downloaded bytes in dry-run mode to determine if any downloading is needed:
```

Comment:
This can also be implemented in `yum` or `dnf` by attempting to run in test-only made with the `--cacheonly` flag to see if it fails due to requesting something to download.

Reply:
I integrated this bit of information into the RFC. Thanks for pointing this out!